### PR TITLE
add box mode convert transform

### LINKED ISSUE #3572: Box transforms
**Is your feature request related to a problem? Please describe.**
Box pre- and post-processing transform functions are not in MONAI

**Describe the solution you'd like**
Add box transform functions

**Describe alternatives you've considered**
N/A

**Additional context**
1. convert input box with original mode to the box with standard mode
2. convert output box with standard mode to the box with original mode
3. Flip boxes with image as a transform

### PR DESCRIPTION
**Pull request by Can-Zhao**
Signed-off-by: Can Zhao <canz@nvidia.com>

Fixes #3572 .

### Description
add box mode convert transform

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/apps/detection/transforms/dictionary.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+        See also :py:class:`monai.apps.detection,transforms.array.BoxConvertToStandard`
+        """
+        super().__init__(box_keys, allow_missing_keys)
+        self.converter = BoxConvertToStandard(mode=mode)
+        self.inverse_converter = BoxConvertMode(src_mode=None, dst_mode=mode)
+
+    def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> Dict[Hashable, NdarrayOrTensor]:
+        d = dict(data)
+        key_id = 0
```
Comment (Nic-Ma) [inline monai/apps/detection/transforms/dictionary.py:135]: May I know what this `key_id` is used for?

Thanks.

File: monai/apps/detection/transforms/dictionary.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        for key in self.key_iterator(d):
+            self.push_transform(d, key)
+            d[key] = self.converter(d[key])
+            key_id += 1
+        return d
+
+    def inverse(self, data: Mapping[Hashable, NdarrayOrTensor]) -> Dict[Hashable, NdarrayOrTensor]:
+        d = deepcopy(dict(data))
+        key_id = 0
```
Comment (Nic-Ma) [inline monai/apps/detection/transforms/dictionary.py:144]: May I know what this `key_id` is used for?

Thanks.

File: tests/test_box_transform.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import unittest
+
+import numpy as np
+from parameterized import parameterized
+
+from monai.apps.detection.transforms.dictionary import BoxConvertToStandardd
+from monai.transforms import Invertd
+from tests.utils import TEST_NDARRAYS, assert_allclose
+
+# this script test box_convert_mode, box_convert_standard_mode
```
Comment (Nic-Ma) [inline tests/test_box_transform.py:21]: Please remove this unrelated comment.

Thanks.

**Review by wyli — APPROVED** [review]
thanks, there's one missing `__init__.py`, I'll add and merge this PR.

**Pull request merged**